File: typewriter-sort.py
# Works only for positive numbers, because, we only have positive numbers in the universe

import random

def typewriter_sort(lst):
    length = 0
    while True:
        length += 1
        num_count = 0

        while True: 
            random_list = [random.randint(0, num_count) for _ in range(length)]
            print(random_list)
            num_count += 1

            if random_list == sorted(lst):
                return random_list

File: typewriter-sort.rs
use rand::Rng;

fn typewriter_sort(lst: &Vec<i32>) -> Vec<i32> {
    let mut length = 0;

    loop {
        length += 1;
        let mut num_count = 0;

        loop {
            let random_list: Vec<i32> = (0..length).map(|_| rand::thread_rng().gen_range(0..=num_count)).collect();
            println!("{:?}", random_list);
            num_count += 1;

            if random_list == lst.iter().copied().collect::<Vec<i32>>().sorted() {
                return random_list;
            }
        }
    }
}